nix/services/mongodb.nix

```diff
@@ -0,0 +1,131 @@
+{ pkgs, lib, name, config, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    package = lib.mkPackageOption pkgs "mongodb" { };
+
+    bind = lib.mkOption {
+      type = types.nullOr types.str;
+      default = "127.0.0.1";
+      description = ''
+        The IP interface to bind to.
+        `null` means "all interfaces".
+      '';
+      example = "127.0.0.1";
+    };
+
+    port = lib.mkOption {
+      type = types.port;
+      default = 27017;
+      description = ''
+        The TCP port to accept connections.
+      '';
+    };
+
+    user = lib.mkOption {
+      type = types.str;
+      default = "default";
+      description = ''
+        The name of the first user to create in
+        Mongo.
+      '';
+      example = "my_user";
+    };
+
+    password = lib.mkOption {
+      type = types.str;
+      default = "password";
+      description = ''
+        The password of the user to configure
+        for initial access.
+      '';
+    };
+
+    extraConfig = lib.mkOption {
+      type = types.lines;
+      default = "";
+      description = "Additional text to be appended to `mongodb.conf`.";
+    };
+  };
+
+  config = {
+    outputs = {
+      settings = {
+        processes = {
+          "${name}" =
+            let
+              mongoConfig = pkgs.writeText "mongodb.conf" ''
+                net.port: ${toString config.port}
+                net.bindIp: ${config.bind}
+                storage.dbPath: ${config.dataDir}
+                ${config.extraConfig}
+              '';
+
+              startScript = pkgs.writeShellApplication {
+                name = "start-mongodb";
+                runtimeInputs = [ pkgs.coreutils config.package ];
+                text = ''
+                  set -euo pipefail
```

Comment:
This line can be removed as `writeShellApplication` sets it by default: https://github.com/NixOS/nixpkgs/blob/0b86640b433332928f978942427300733ebca24c/pkgs/build-support/trivial-builders/default.nix#L253C23-L253C53

Reply:
> This line can be removed

I am referring to `set -euo pipefail`

Reply:
Removed.